Firestore/Source/Model/FSTFieldValue.mm

```diff
@@ -47,28 +47,47 @@
 
 @implementation FSTFieldValueOptions
 
-+ (instancetype)optionsForSnapshotOptions:(FIRSnapshotOptions *)options {
-  if (options.serverTimestampBehavior == FSTServerTimestampBehaviorNone) {
-    static FSTFieldValueOptions *defaultInstance = nil;
-    static dispatch_once_t onceToken;
```

Comment:
Since you mentioned in Android review that default options seem unnecessary, perhaps I can delete them?

Reply:
@wilhuff, please let me know if you think this should be kept.

Reply:
Doesn't matter much to me. Doing it the old way saves a heap allocation.

Once tokens are going away anyway though so I wouldn't stress about this:
  * Most of our types are migrating to value types
  * Simple value types don't require a heap allocation so we wouldn't be saving anything necessarily by sharing
  * Magic statics make this work otherwise if there's benefit in sharing